### PR TITLE
Added Pythagoras Theorem

### PR DESCRIPTION
**Pull request by localleon**
Calculates the distance between to vectors/points in space with the Pythagoras Theorem.

### CODE REVIEW

**Comment by vedantmamgain** [comment]
Remove the main function.

**Comment by localleon** [comment]
> Remove the main function.
Done. Also renamed the package to make it more universal

File: math/pythagoras/pythagoras.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	x int
+	y int
+	z int
+}
+
+//Distance calculates the distance between to vectors with the   Pythagoras theorem
+func Distance(a, b Vector) float64 {
+	abs := float64(math.Abs(float64(b.x - a.x)))
+	res := float64(math.Pow(abs, 2.0) + math.Pow(float64(b.y-a.y), 2.0) + math.Pow(float64(b.z-a.z), 2))
+	fmt.Println(abs, res)
```
Comment (Mystigan) [inline math/pythagoras/pythagoras.go:20]: Please remove this print statement.

**Review by Mystigan — CHANGES_REQUESTED** [review]
- Please remove the print statement.
- Let's keep the power factor consistent (Line 19: currently it's 2.0 for x and y components and 2 for z).
- Please add some more test cases and consider using table driven tests.( Refer: https://github.com/golang/go/wiki/TableDrivenTests )

File: math/pythagoras/pythagoras.go (hunk 1 above)
Comment (localleon) [reply math/pythagoras/pythagoras.go:20]: Sure thing.

**Comment by localleon** [comment]
>     * Please remove the print statement.
> 
>     * Let's keep the power factor consistent (Line 19: currently it's 2.0 for x and y components and 2 for z).
> 
>     * Please add some more test cases and consider using table driven tests.( Refer: https://github.com/golang/go/wiki/TableDrivenTests )

Everything should be fixed now. I needed to round the results of the Distance Function for the tests because i wasn't able to find nice number for round results and i also wanted to test edge cases. Writing down the whole 64bit of the float number would be kinda difficult to do in a testfile

File: math/pythagoras/pythagoras.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//Vector defines a tuple with 3 values in 3d-space
+type Vector struct {
+	x float64
+	y float64
+	z float64
+}
+
+//Distance calculates the distance between to vectors with the   Pythagoras theorem
+func Distance(a, b Vector) float64 {
+	abs := float64(math.Abs(b.x - a.x))
```
Comment (Mystigan) [inline math/pythagoras/pythagoras.go:16]: Why are calculating absolute value for the x component. We are guaranteed to get a positive number when we square it.

File: math/pythagoras/pythagoras_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	{Vector{2, -1, 7}, Vector{1, -3, 5}, 3.0},
+	{Vector{4, 10, 9}, Vector{4, 3, 5}, 8.06},
+	{Vector{8, 5, 5}, Vector{1, 1, 12}, 10.67},
+	{Vector{1, 1, 1}, Vector{2, 2, 2}, 1.73},
+}
+
+//TestDistance tests the Function Distance with 2 vectors
+func TestDistance(t *testing.T) {
+	for _, tt := range distanceTest {
+		res := Distance(tt.v1, tt.v2)           // Calculate result for
```
Comment (Mystigan) [inline math/pythagoras/pythagoras_test.go:23]: Please enclose each iteration in a t.Run() func. This would ensure that even if one of the test cases fail, the others will still be executed. The func will require a name for the test case which you can add as a part of distanceTest slice. You may use [this test file](math/primecheck/prime_check_test.go) for  reference.

**Review by Mystigan — CHANGES_REQUESTED** [review]
Please work on the suggested changes and we should be ready to go.

**Review by Mystigan — APPROVED** [review]
LGTM. Thanks for your contribution!
